### ISSUE TITLE
Custom encoding for tweaked public keys in Taproot

### ISSUE THREAD

**Issue by dr-orlovsky** [rust-bitcoin/rust-bitcoin#737]
Background: https://github.com/rust-bitcoin/rust-bitcoin/pull/721#discussion_r761390759

I personally prefer to use some form of bech32m encoding with custom HRP

**Comment by Kixunil** [comment]
Would it make sense to standardize this? Talk with core/other wallet devs and make it a (small) BIP.

**Comment by apoelstra** [comment]
There should basically be no untweaked public keys out there in the wild. We have the `rawtr` descriptor in Core for them. Do we need more?

**Comment by Kixunil** [comment]
@apoelstra do you mean tweaked?

**Comment by apoelstra** [comment]
No, I mean untweaked. Pretty-much every Taproot address should have a tweaked key.

**Comment by sanket1729** [comment]
Just recapping, the internal key is `UnTweakedPubKey` and the output key is `TweakedPubKey`. 
- For the wallet descriptors we should use `tr(untweaked_key)`, 
- For bech32m address purposes, we only need to encode `TweakedPubkey` as it is the one that corresponds to the address. 

So, TweakedPubkeys are already encoded as addresses, do you need something else?

**Comment by Kixunil** [comment]
@sanket1729 binary encoding may be a good idea to reduce bandwidth and maybe somehow distinguish untweaked from tweaked keys.

**Comment by dr-orlovsky** [comment]
It is not clear where this binary encoding can be used though. For network serialization we follow consensus encoding rules; for serde, we are bound by the data type.

I propose to close this issue since the way it is currently done for me seems to be the only possible way.

**Comment by Kixunil** [comment]
> we are bound by the data type

I don't understand what do you mean by this, can you explain?

**Comment by dr-orlovsky** [comment]
I mean that serde deserializes into a specific data type which is declared in the field itself - and additional/custom encoding is not needed

**Comment by Kixunil** [comment]
Oh, you mean derive. There's a bunch of ways to customize that or even skip derive entirely and write it by hand.

**Comment by dr-orlovsky** [comment]
@Kixunil so can we close this issue or there is still a plan to do some custom binary encoding?

**Comment by Kixunil** [comment]
Let's close this and if anyone needs efficient binary encoding we can reopen. There shouldn't be any problem adding it later and I don't think it's urgent for anyone.
